### PR TITLE
DRY docker shell scripts for use with docker CI

### PR DESCRIPTION
**Pull request by andrewvc**
We'll be using ci/docker_run.sh to run our x-pack build scripts soon,
so this refactor makes sense

### CODE REVIEW

File: ci/docker_run.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    # Jenkins
+    IMAGE_NAME=$branch_specifier"-"$(date +%s%N)
+fi
+
+echo "Running Docker CI build for '$IMAGE_NAME' "
+
+docker build  -t $IMAGE_NAME .
+exit_code=$?; [[ $exit_code != 0 ]] && exit $exit_code
+# Run the command, skip the first argument, which is the local script name
+docker run --sig-proxy=true --rm $IMAGE_NAME ${@:2}
```
Comment (jakelandis) [inline ci/docker_run.sh]: does this skip the first command , or just execute the second command ? 
e.g.
for these arguments : `a b c d` will this result in `b c d` or just `b` ?

Comment (andrewvc) [reply ci/docker_run.sh]: Skips the first one, includes the rest. So, `b c d`. The first argument is the docker image name of course

Comment (andrewvc) [reply ci/docker_run.sh]: I see my comment in the code is incorrect, it says it skips the script name, it actually skips the image name

**Comment by jakelandis** [comment]
LGTM

**Pull request closed without merging**